### PR TITLE
Use color property whitelist

### LINKED ISSUE #217: Apple Pay styling should not be treated as color
Safari has special vendored CSS properties for Apple Pay button styling: https://developer.apple.com/documentation/apple_pay_on_the_web/displaying_apple_pay_buttons

As seen from the docs, one of these properties might look like following:

```css
    .apple-pay-button-black {
        -apple-pay-button-style: black;
    }
    .apple-pay-button-white {
        -apple-pay-button-style: white;
    }
    .apple-pay-button-white-with-line {
        -apple-pay-button-style: white-outline;
    }
```

Currently minifier assumes that `black` and `white` are regular CSS colors, no matter what property they're in, and minifies them to `#000` and `#fff` correspondingly, which break styling, as for Safari these are expected to be regular names.

This could be treated as a browser-specific quirks and disabled just for that particular property, but blacklisting doesn't like a very failproof solution, since there are other contexts where regular identifiers might potentially have name collisions with colors, but shouldn't be treated as such.

### PR DESCRIPTION
**Pull request by tdewolff**
Fixes #217

Uses a whitelist to employ color transformations (name/hex to hex). Probably missing some properties in the whitelist, need to do some tests on large files.

### CODE REVIEW

File: css/css.go [hunk 1]
```diff
@@ -867,40 +867,44 @@ func (c *cssMinifier) shortenToken(prop css.Hash, tt css.TokenType, data []byte)
...
+			}
+			if ident, ok := ShortenColorHex[string(data)]; ok {
+				tt = css.IdentToken
+				data = ident
+			} else if len(data) == 7 && data[1] == data[2] && data[3] == data[4] && data[5] == data[6] {
+				tt = css.HashToken
+				data[2] = data[3]
+				data[3] = data[5]
+				data = data[:4]
+			} else if len(data) == 9 && data[1] == data[2] && data[3] == data[4] && data[5] == data[6] && data[7] == data[8] {
```
Comment (RReverser) [inline css/css.go]: Wonder if this condition could/should be somehow nested into previous as most conditions and operations are shared.

File: css/css_test.go [hunk 2]
```diff
@@ -155,7 +155,7 @@ func TestCSSInline(t *testing.T) {
 		{"content: \"a\\\nb\";", "content:\"ab\""},
 		{"content: \"a\\\r\nb\\\r\nc\";", "content:\"abc\""},
 		{"content: \"\";", "content:\"\""},
-		{"x: white , white", "x:#fff,#fff"},
+		{"color: white , white", "color:#fff,#fff"},
```
Comment (RReverser) [inline css/css_test.go]: Could you add some tests that make sure color-like identifier are preserved in other contexts? (e.g. Apple Pay from original report)

Comment (RReverser) [reply css/css_test.go]: Same for KeepCSS2 tests.

**Comment by tdewolff** [comment]
Not sure if this PR moves into the right direction. We probably want better handling of value types (color, time, length, ...) depending on the declaration name in general, not just for color. White listing an entire declaration dictates nothing about the specific values it may have (maybe the first value is a color but the second is not). See #234 on plans to handling different value types.

**Pull request closed without merging**